autoload/neomake/makers/clippy.vim

```diff
@@ -0,0 +1,38 @@
+" vim: ts=4 sw=4 et
+
+function! neomake#makers#clippy#clippy() abort
+    let errorfmt = '%Eerror[E%n]: %m,'.
+                 \ '%Eerror: %m,'.
+                 \ '%Wwarning: %m,'.
+                 \ '%Inote: %m,'.
+                 \ '%-Z\ %#-->\ %f:%l:%c,'.
+                 \ '%I\ %#\= %t%*[^:]: %m,'.
+                 \ '%I\ %#|\ %#%\\^%\\+ %m,'.
+                 \ '%-G%s,'
+
+    " When rustup and a nightly toolchain is installed, that is used.
+    " Otherwise, the default cargo exectuable is used. If this is not part
+    " of a nightly rust, this will fail.
+    if !exists('s:rustup_has_nightly')
+        if !executable('rustup')
+            let s:rustup_has_nightly = 0
+        else
+            call system('rustup show | grep -q "^nightly-"')
+            let s:rustup_has_nightly = !v:shell_error
+        endif
+    endif
```

Comment:
Very good.
An improvement would be to set it to -1 in the outer scope, which allows to refresh the state when the script is resourced (then it gets set to -1 again).  Currently it will be set always.  Sorry for not suggesting this right away.
(I often re-source files, e.g. specific makers while changing them using `:Runtime` from https://github.com/tpope/vim-scriptease)